config/routes.rb

```diff
@@ -26,6 +27,12 @@
   curation_concerns_basic_routes
   concern :exportable, Blacklight::Routes::Exportable.new
 
+  Hyrax::Engine.routes.draw do 
+    get '/xmlui/handle/:handle_id/:id', to: 'handles#handle_show', as: 'handle_basic_show'
+    get '/xmlui/handle/:handle_id/:id/(:show)', to: 'handles#handle_show', as: 'handle_specific_show'
```

Comment:
routes with optional params (like you saw on the example urls, no doubt) aren't necessary. I think, in general, we'd just want to redirect people to the show page of the work that download could be found in (or the show page for a regular link)

You don't need specify the querystring params: http://guides.rubyonrails.org/routing.html#the-query-string